lib/test_queue/iterator.rb

```diff
@@ -37,7 +37,8 @@ def each
           else
             yield suite
           end
-          @stats[suite.to_s] = Time.now - start
+          key = suite.respond_to?(:id) ? suite.id : suite
```

Comment:
I think this is missing a `.to_s`?

`key = suite.respond_to?(:id) ? suite.id : suite.to_s`